### PR TITLE
Upgrade Helm for connectedk8s

### PR DESCRIPTION
**Pull request by kitnhl**
Added changes necessary to upgrade Helm used in connectedk8s extension from Helm v3.6.3 to Helm v3.12.2.

image

The resulting functionality matrix can be seen above there, where `->-> Agent vX` means updating to version `X` and the `*` indicates that this operation installs Agents of the desired version but in older packaging.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by kitnhl** [comment]
Do not merge until Agents v1.14 are fully released.

**Comment by kitnhl** [comment]
> Thank you for your contribution! We will review the pull request and get back to you soon.

Hi @yonzhan and team, could I get a review for this PR please?

File: src/connectedk8s/azext_connectedk8s/_precheckutils.py
Comment (NarayanThiru) [inline src/connectedk8s/azext_connectedk8s/_precheckutils.py:139]: Why are you explicitly passing new_path as False?

**Review by NarayanThiru — CHANGES_REQUESTED** [review]

**Comment by yanzhudd** [comment]
Hi @kitnhl,
please note that CLI only reviews the code style and specifications but not responsible for business logic, so it is better to involve service team to review it as well. After it is approved from service team, we would start to review this PR.

**Comment by yanzhudd** [comment]
Please resolve CI failed items.

File: src/connectedk8s/azext_connectedk8s/_precheckutils.py
Comment (kitnhl) [reply src/connectedk8s/azext_connectedk8s/_precheckutils.py:139]: The precheck pulls from `consts.Cluster_Diagnostic_Checks_Job_Registry_Path` defined as `mcr.microsoft.com/azurearck8s/helmchart/stable/clusterdiagnosticchecks:0.1.2` which does not have a newly packaged path/version. I actually have no clue in which pipeline/by which Helm version did this chart get packaged. Because the prechecks succeed fine, I can, however, imply that it was done using a new enough Helm version because it can be pulled using Helm 3.12 just like all the newly packaged stuff.

Comment (NarayanThiru) [reply src/connectedk8s/azext_connectedk8s/_precheckutils.py:139]: So, basically your assumption is that for create, update, upgrade-agents, enable-features and disable-features and disable-cluster-connect, you are checking the min. version and for pre-check you are letting it go without this check.

While this may be fine, I think it is good to confirm why pre-check works.  I believe Abhilash might have some knowledge in this area.  Can you check with him to confirm?

The reason I am asking is because, since we are checking for min version in pre-check, I wonder if may fail in the future and cause CRIs.

Comment (kitnhl) [reply src/connectedk8s/azext_connectedk8s/_precheckutils.py:139]: Ahhh sorry I might need to explain more. The version check there is for the version of the chart being pulled, and for everything but precheck, we want that (dynamically specified) chart version to be greater than or equal to `1.14`. The `clusterdiagnosticchecks` chart path and version are constant as I shared above, namely `mcr.microsoft.com/azurearck8s/helmchart/stable/clusterdiagnosticchecks:0.1.2` i.e. version `0.1.2` and it would be no use to check this constant. Does that make more sense? Will check with Abhilash to see if he knows anything about this particular chart anyways!

**Review by NarayanThiru — APPROVED** [review]
:shipit:

**Review by yanzhudd — APPROVED** [review]

**Comment by kitnhl** [comment]
@yanzhudd thank you, should be good for you guys to take a look now :)

**Pull request merged**
